modules/data.land/R/soil_utils.R

```diff
@@ -0,0 +1,266 @@
+  sand <- 100. * sandfrac
+  clay <- 100. * clayfrac
+  silt <- 100. - sand - clay
+  #---------------------------------------------------------------------------------------#
+  
+  #---------------------------------------------------------------------------------------#
+  #     Here there is not much we can do other than explore where in the triangle space   #
+  # we are.                                                                               #
+  #---------------------------------------------------------------------------------------#
+  
+  if (silt > 100. | silt < 0. | sand > 100. | sand < 0. | clay > 100. | clay < 0. ) {
+    print("---------------------------------------------------")
+    print(" At least one of your percentages is screwy...")
+    print(paste("SAND <- ",sprintf("%.2f",sand),"%",sep=""))
+    print(paste("CLAY <- ",sprintf("%.2f",clay),"%",sep=""))
+    print(paste("SILT <- ",sprintf("%.2f",silt),"%",sep=""))
+    print("---------------------------------------------------")
+    stop ("This soil doesn''t fit into any category...")
+    
+  }else if(sand > 85.0 + 0.5 * clay) {
+    mysoil <-  1 #----- Sand. ------------------------------------------------------------#
+  }else if(sand > 70.0 + clay) {
+    mysoil <-  2 #----- Loamy sand. ------------------------------------------------------#
+  }else if((clay <= 20.0 & sand > 52.5) | (clay <= 7.5 & silt <= 50.0)) {
+    mysoil <-  3 #----- Sandy loam. ------------------------------------------------------#
+  }else if((clay <= 27.5 & silt > 50.0 & silt <= 80.0) | (silt >  80.0 & clay > 12.5)) {
+    mysoil <-  4 #----- Silt loam. -------------------------------------------------------#
+  }else if(clay > 7.5 & clay <= 27.5 & silt > 27.5 & silt <= 50.0 & sand <= 52.5) {
+    mysoil <-  5 #----- Loam. ------------------------------------------------------------#
+  }else if(clay > 20.0 & clay <= 35.0 & silt <= 27.5 & sand > 45.0) {
+    mysoil <-  6 #----- Sandy clay loam. -------------------------------------------------#
+  }else if(clay > 27.5 & clay <= 40.0 & sand <= 20.0) {
+    mysoil <-  7 #----- Silty clay loam. -------------------------------------------------#
+  }else if(clay > 27.5 & clay <= 40.0 & sand > 20.0 & sand <= 45.0) {
+    mysoil <-  8 #----- Clayey loam. -----------------------------------------------------#
+  }else if(clay > 35.0 & sand > 45.0) {
+    mysoil <-  9 #----- Sandy clay. ------------------------------------------------------#
+  }else if(clay > 40.0 & silt > 40.0) {
+    mysoil <- 10 #----- Silty clay. ------------------------------------------------------#
+  }else if(clay <= 70.0 & sand <= 30.0 & silt <= 30.0) {
+    mysoil <- 11 #----- Clay. ------------------------------------------------------------#
+  }else if( silt > 80.0 & clay <= 12.5) {
+    mysoil <- 14 #----- Silt. ------------------------------------------------------------#
```

Comment:
@ankurdesai Unfortunately  my understanding is that all these texture <> class relationships really only hold for mineral soil, so there's no sure way to separate peat from silt if you only have particle size (sand/silt/clay) data. Though of course If information about organic matter content *is* available, we could provide a separate function that acts upstream of this one?